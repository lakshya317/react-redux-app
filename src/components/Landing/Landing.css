.landing {
    height: 100vh;
    width: 100vw;
    display: flex;
    flex-direction: column;
    font-family: 'Verdana';
}

@font-face {
    font-family: 'Verdana';
    src: local('Verdana'), url('../../assets/fonts/Verdana.ttf') format('truetype');
}

.landing-body {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    flex: 1 1 auto;
}

.landing-title {
    text-align: center;
    font-size: 30px;
    font-weight: 700;
}

.stack-container {
    width: 30%;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
}

.stack-group {
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.stack-title {
    text-align: center;
    font-weight: 600;
}

.login-button-container {
    display: flex;
    justify-content: center;
    flex-grow: 0.4;
}

.login-link {
    align-self: flex-start;
}

.login-link-button {
    color: white;
    background-color: var(--shell-red);
    font-weight: 600;
    padding: 5px 10px;
    border: 2px solid var(--shell-red);
    border-radius: 4px;
}

.no-tooltips {
    pointer-events: none;
}
