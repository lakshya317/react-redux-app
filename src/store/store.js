import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import { encryptTransform } from 'redux-persist-transform-encrypt';
import storage from 'redux-persist/lib/storage';
import reducer from '../reducer';
import middleware from '../middleware';

// Config
const persistStoreEncryptionKey = process.env.REACT_APP_PERSIST_STORE_ENCRYPTION_KEY;
const persistConfigSecret = process.env.REACT_APP_PERSIST_CONFIG_SECRET;

//Encryptor
const encryptor = encryptTransform({
    secretKey: persistStoreEncryptionKey,
    onError: function (error) {
        // Handle the error.
        console.error(error);
    }
});

const persistConfig = {
    transforms: [encryptor],
    storage,
    key: persistConfigSecret
};

const persistedReducer = persistReducer(persistConfig, reducer);

export const store = createStore(persistedReducer, middleware);
export const persistor = persistStore(store);
