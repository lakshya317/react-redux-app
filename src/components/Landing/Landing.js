import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import './Landing.css'
import Header from "../Header/Header"
import { ReactComponent as ReactLogo } from '../../images/React.svg';
import { ReactComponent as ReduxLogo } from '../../images/Redux.svg';
import ErrorModal from '../Modal/ErrorModal';
import { removeDisplayError } from "../../action/displayError"


const Landing = (props) => {

    //Redux State
    const displayError = useSelector((state) => state.displayError)

    //dispatch
    const dispatch = useDispatch()

    return (
        <div className='landing'>
            <ErrorModal errormessage={displayError} onHide={() => {dispatch(removeDisplayError());}} />
            <Header />
            <div className='landing-body'>
                <div className='landing-title'>
                    Welcome to a Login Screen made using
                </div>
                <div className="stack-container">
                    <div className='stack-group'>
                        <ReactLogo height={"100px"} width={"100px"} title={"React Logo"}/>
                        <div className="stack-title">
                            React
                        </div>
                    </div>
                    <div className='stack-group'>
                        <ReduxLogo height={"100px"} width={"100px"} title={"ReduxLogo"}/>
                        <div className="stack-title">
                            Redux
                        </div>
                    </div>
                </div>
                <div className="login-button-container">
                    <Link className='login-link' to="/login" style={{ textDecoration: 'none' }}>
                        <div className='login-link-button'>
                            Login
                        </div>
                    </Link>
                </div>
            </div>
        </div>
    );
}

export default Landing;
