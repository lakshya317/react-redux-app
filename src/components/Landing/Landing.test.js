import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from "../../store/store";
import Landing from './Landing';

describe('Landing Page Tests', () => {

    test('Landing page snapshot', () => {
        
        const landing = render(
            <Provider store={store}>
                <MemoryRouter>
                    <Landing />
                </MemoryRouter>
            </Provider>);
        
        expect(landing).toMatchSnapshot()
    });

    

})