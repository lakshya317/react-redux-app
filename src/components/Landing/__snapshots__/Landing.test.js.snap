// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Tests Landing page snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="landing"
      >
        <div
          class="header"
        >
          <a
            href="/"
            style="text-decoration: none;"
          >
            <img
              alt="Shell Logo"
              class="Shell-logo"
              src="Shell_Logo.png"
            />
          </a>
        </div>
        <div
          class="landing-body"
        >
          <div
            class="landing-title"
          >
            Welcome to an Application made using:
          </div>
          <div
            class="stack-container"
          >
            <div
              class="stack-group"
            >
              <svg
                class="no-tooltips"
                height="100px"
                title="React Logo"
                width="100px"
              >
                React.svg
              </svg>
              <div
                class="stack-title"
              >
                React
              </div>
            </div>
            <div
              class="stack-group"
            >
              <svg
                class="no-tooltips"
                height="100px"
                title="Redux Logo"
                width="100px"
              >
                Redux.svg
              </svg>
              <div
                class="stack-title"
              >
                Redux
              </div>
            </div>
          </div>
          <div
            class="login-button-container"
          >
            <a
              class="login-link"
              href="/login"
              style="text-decoration: none;"
            >
              <div
                class="login-link-button"
              >
                Login
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="landing"
    >
      <div
        class="header"
      >
        <a
          href="/"
          style="text-decoration: none;"
        >
          <img
            alt="Shell Logo"
            class="Shell-logo"
            src="Shell_Logo.png"
          />
        </a>
      </div>
      <div
        class="landing-body"
      >
        <div
          class="landing-title"
        >
          Welcome to an Application made using:
        </div>
        <div
          class="stack-container"
        >
          <div
            class="stack-group"
          >
            <svg
              class="no-tooltips"
              height="100px"
              title="React Logo"
              width="100px"
            >
              React.svg
            </svg>
            <div
              class="stack-title"
            >
              React
            </div>
          </div>
          <div
            class="stack-group"
          >
            <svg
              class="no-tooltips"
              height="100px"
              title="Redux Logo"
              width="100px"
            >
              Redux.svg
            </svg>
            <div
              class="stack-title"
            >
              Redux
            </div>
          </div>
        </div>
        <div
          class="login-button-container"
        >
          <a
            class="login-link"
            href="/login"
            style="text-decoration: none;"
          >
            <div
              class="login-link-button"
            >
              Login
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
